waveform_editor/configuration.py

```diff
@@ -115,9 +138,24 @@ def replace_waveform(self, waveform):
             raise ValueError(
                 f"Waveform '{waveform.name}' does not exist in the configuration."
             )
-        else:
-            group = self.waveform_map[waveform.name]
-            group.waveforms[waveform.name] = waveform
+
+        group = self.waveform_map[waveform.name]
+        old_waveform = group.waveforms[waveform.name]
+
+        group.waveforms[waveform.name] = waveform
+
+        try:
+            if isinstance(waveform, DerivedWaveform):
```

Comment:
Can we check in the editor whether the DerivedWaveform that you're editing is circular and avoid calling this method in that case altogether? That would make this method a lot easier and avoid al this error handling and reverting?